pyomo/contrib/doe/fim_doe.py

```diff
@@ -0,0 +1,2582 @@
+                jac[para] = list_jac
+
+        # check if another prior experiment FIM is provided other than the user-specified one
+        if specified_prior is None:
+            prior_in_use = self.prior_FIM
+        else:
+            prior_in_use = specified_prior
+
+        # Assemble and analyze results
+        FIM_analysis = FIM_result(self.param_name, self.measure, jacobian_info=None, all_jacobian_info=jac,
+                                    prior_FIM=prior_in_use, store_FIM=FIM_store_name, scale_constant_value=self.scale_constant_value)
+
+        time11 = time.time()
+        self.logger.info('Build time with sequential_sipopt or kaug mode [s]:', sum(time_allbuild))
+        self.logger.info('Solve time with sequential_sipopt or kaug mode [s]:', sum(time_allsolve))
+        self.logger.info('Total wall clock time [s]:', time11-time00)
+
+        self.jac = jac
+        FIM_analysis.build_time = sum(time_allbuild)
+        FIM_analysis.solve_time = sum(time_allsolve)
+
+        return FIM_analysis
+
+    def __direct_kaug(self):
+        time00 = time.time()
+        # create scenario class for a base case
+        scena_gen = Scenario_generator(self.param_init, formula=None, step=self.step)
+        scenario_all = scena_gen.simultaneous_scenario()
+
+        # create model
+        time0_build = time.time()
+        mod = self.create_model(scenario_all, args=self.args)
+        time1_build = time.time()
+        time_build = time1_build - time0_build
+
+        # discretize if needed
+        if self.discretize_model is not None:
+            mod = self.discretize_model(mod)
+
+        # get all time
+        t_all = []
+        for t in mod.t:
+            t_all.append(t)
+
+        # add objective function
+        mod.Obj = pyo.Objective(expr=0, sense=pyo.minimize)
+
+        # Check if measurement time points are in this time set
+        # Also correct the measurement time points
+        # For e.g. if a measurement time point is 0.0 in the model but is given as 0, it is corrected here
+        measurement_accurate_time = self.flatten_measure_timeset.copy()
+        for j in self.flatten_measure_name:
+            for no_t, tt in enumerate(self.flatten_measure_timeset[j]):
+                if tt not in t_all:
+                    self.logger.warning('A measurement time point not measured by this model: ', tt)
+                else:
+                    measurement_accurate_time[j][no_t] = t_all[t_all.index(tt)]
+
+        # set ub and lb to parameters
+        for par in self.param_name:
+            component = getattr(mod, par)[0]
+            component.setlb(self.param_init[par])
+            component.setub(self.param_init[par])
+
+        # generate parameter name list and value dictionary with index
+        var_name = []
+        var_dict = {}
+        for name in self.param_name:
+            var_name.append(name+'[0]')
+            var_dict[name+'[0]'] = self.param_init[name]
```

Comment:
What is significant about the 0th index?  This should be documented (at a minimum), and preferably should not be hard-coded